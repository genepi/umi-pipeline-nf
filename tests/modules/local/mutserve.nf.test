nextflow_process {

    name "Test Process MUTSERVE"
    script "modules/local/variant_calling/mutserve.nf"
    process "MUTSERVE"

    test("Should run without failures") {

        when {
            params {
                variant_calling = true 
                variant_caller = "mutserve"
            }
            process {
                """
                input[0] = Channel.of(["lpa2645", file("$baseDir/tests/input/pipeline/ref/lpa-ref2645.bed"), "barcode03", "$baseDir/tests/input/variant_calling/barcode03/final.bam", "$baseDir/tests/input/variant_calling/barcode03/final.bam.bai"], )
                input[1] = "final"
                input[2] = file("$baseDir/tests/input/pipeline/ref/lpa-ref2645.fasta")
                input[3] = file("$baseDir/tests/input/pipeline/ref/lpa-ref2645.fasta.fai")
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }
    }

    test("Should output three files") {

        when {
            params {
                variant_calling = true 
                variant_caller = "mutserve"
            }
            process {
                """
                input[0] = Channel.of(
                    ["lpa2645", file("$baseDir/tests/input/pipeline/ref/lpa-ref2645.bed"), "barcode03", "$baseDir/tests/input/variant_calling/barcode03/final.bam", "$baseDir/tests/input/variant_calling/barcode03/final.bam.bai"],
                    ["lpa2645", file("$baseDir/tests/input/pipeline/ref/lpa-ref2645.bed"), "barcode04", "$baseDir/tests/input/variant_calling/barcode04/final.bam", "$baseDir/tests/input/variant_calling/barcode04/final.bam.bai"],
                    ["lpa2645", file("$baseDir/tests/input/pipeline/ref/lpa-ref2645.bed"), "barcode05", "$baseDir/tests/input/variant_calling/barcode05/final.bam", "$baseDir/tests/input/variant_calling/barcode05/final.bam.bai"]
                    )
                input[1] = "final"
                input[2] = file("$baseDir/tests/input/pipeline/ref/lpa-ref2645.fasta")
                input[3] = file("$baseDir/tests/input/pipeline/ref/lpa-ref2645.fasta.fai")
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }
}
