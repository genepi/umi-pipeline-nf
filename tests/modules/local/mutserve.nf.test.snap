{
    "Should output three files": {
        "content": [
            {
                "0": [
                    "final.txt:md5,1e5554e27b3b8a0687374df975a6c9cb",
                    "final.txt:md5,871ecfddc44cb8b3aee5a3c482fa2900",
                    "final.txt:md5,bdcd9fa23bcd1e226b07988460f9b162"
                ],
                "1": [
                    "final_raw.txt:md5,94352f26d9095f38eb44ff730db75dbc",
                    "final_raw.txt:md5,a647c13b88e318322308fa70526a8d96",
                    "final_raw.txt:md5,f8da170e2122c7caffa8fb3efb311434"
                ],
                "2": [
                    "final_parsed.txt:md5,1299a3ffd233f79210415a5c4d472eea",
                    "final_parsed.txt:md5,6d367fbfc32f86586b02ed68c1bad36b",
                    "final_parsed.txt:md5,d6644ef51496cf09047606032cbdab63"
                ],
                "variants": [
                    "final.txt:md5,1e5554e27b3b8a0687374df975a6c9cb",
                    "final.txt:md5,871ecfddc44cb8b3aee5a3c482fa2900",
                    "final.txt:md5,bdcd9fa23bcd1e226b07988460f9b162"
                ],
                "variants_parsed": [
                    "final_parsed.txt:md5,1299a3ffd233f79210415a5c4d472eea",
                    "final_parsed.txt:md5,6d367fbfc32f86586b02ed68c1bad36b",
                    "final_parsed.txt:md5,d6644ef51496cf09047606032cbdab63"
                ],
                "variants_raw": [
                    "final_raw.txt:md5,94352f26d9095f38eb44ff730db75dbc",
                    "final_raw.txt:md5,a647c13b88e318322308fa70526a8d96",
                    "final_raw.txt:md5,f8da170e2122c7caffa8fb3efb311434"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0-rc2",
            "nextflow": "24.10.0"
        },
        "timestamp": "2025-05-15T17:07:23.870331577"
    },
    "Should run without failures": {
        "content": [
            {
                "0": [
                    "final.txt:md5,1e5554e27b3b8a0687374df975a6c9cb"
                ],
                "1": [
                    "final_raw.txt:md5,94352f26d9095f38eb44ff730db75dbc"
                ],
                "2": [
                    "final_parsed.txt:md5,6d367fbfc32f86586b02ed68c1bad36b"
                ],
                "variants": [
                    "final.txt:md5,1e5554e27b3b8a0687374df975a6c9cb"
                ],
                "variants_parsed": [
                    "final_parsed.txt:md5,6d367fbfc32f86586b02ed68c1bad36b"
                ],
                "variants_raw": [
                    "final_raw.txt:md5,94352f26d9095f38eb44ff730db75dbc"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0-rc2",
            "nextflow": "24.10.0"
        },
        "timestamp": "2025-05-15T17:07:12.453706442"
    }
}